opsdroid/connector/matrix/connector.py

```diff
@@ -405,11 +404,14 @@ async def _file_to_mxc_url(self, file_event):
                 mxc_url = file_event.url
 
         if not mxc_url:
+            upload_file = await file_event.get_file_bytes()
+            if isinstance(upload_file, bytes):
+                upload_file = BufferedReader(BytesIO(upload_file))
```

Comment:
I am not very familiar with the `io` classes, but what is `BufferedReader` providing here? Am I correct in thinking that `BytesIO` already provides a file-like input and that's all we need to pass to `nio`?

Reply:
BytesIO is a file-like but the upload function expects `TextIOBase, BufferedReader, BufferedRandom, FileIO, AsyncBufferedReader or AsyncTextIOWrapper` and `bytes` cant be directly casted to these types

Reply:
Oh, what types it actually supports in `io` are really bizzare to me, I might ask in the nio room.

It might actually be easier to pass it a callable, which returns the bytes directly, I think this will work: `upload_file = lambda x, y: upload_file`.